spikeextractors/extractors/nwbextractors/nwbextractors.py

```diff
@@ -686,61 +688,66 @@ def add_electrical_series(
-                    LFPElectricalSeries=dict(lfp_defaults)
-                )
-        assert ('ElectricalSeries' in metadata['Ecephys']
-                and isinstance(metadata['Ecephys']['ElectricalSeries'], dict)) \
-               or ('LFPElectricalSeries' in metadata['Ecephys']
-                   and isinstance(metadata['Ecephys']['LFPElectricalSeries'], dict)), \
-               ("Expected either metadata['Ecephys']['ElectricalSeries'] or "
-                "metadata['Ecephys']['LFPElectricalSeries'] to be a dictionary!")
-
-        if not write_as_lfp:
-            es_name = metadata['Ecephys']['ElectricalSeries'].get('name', raw_defaults['name'])
-            assert es_name not in nwbfile.acquisition, \
-                f"Raw ElectricalSeries '{es_name}' is already written in the NWBFile!"
-        else:
-            es_name = metadata['Ecephys']['LFPElectricalSeries'].get('name', lfp_defaults['name'])
+        assert write_as in ['raw', 'processed', 'lfp'], \
+            f"'write_as' should be 'raw', 'processed' or 'lfp', but intead received value {write_as}"
+        if write_as == 'raw':
+            eseries_kwargs = dict(
+                name="ElectricalSeries_raw",
+                description="Raw acquired data",
+                comments="Generated from SpikeInterface::NwbRecordingExtractor"
+            )
+        elif write_as == 'processed':
+            eseries_kwargs = dict(
+                name="ElectricalSeries_processed",
+                description="Processed data",
+                comments="Generated from SpikeInterface::NwbRecordingExtractor"
+            )
+        elif write_as == 'lfp':
+            eseries_kwargs = dict(
+                name="ElectricalSeries_lfp",
+                description="Processed data - LFP",
+                comments="Generated from SpikeInterface::NwbRecordingExtractor"
+            )
+
+        # If user passed metadata info, overwrite defaults
+        if es_key is not None and metadata is not None:
+            assert es_key in metadata.get('Ecephys', dict()).keys(), f"Metadata dictionary does not contain key '{es_key}'"
+            eseries_kwargs.update(metadata['Ecephys'][es_key])
+
+        # Check for existing names and processing modules in nwbfile
+        if write_as == 'raw':
+            assert eseries_kwargs['name'] not in nwbfile.acquisition, f"Raw ElectricalSeries '{es_name}' is already written in the NWBFile!"
+        elif write_as == 'processed':
             ecephys_mod = check_module(
-                nwbfile,
-                'ecephys',
-                "Intermediate data from extracellular electrophysiology recordings, e.g., LFP."
+                nwbfile=nwbfile,
+                name='ecephys',
+                description="Intermediate data from extracellular electrophysiology recordings, e.g., LFP."
+            )
+            if 'Processed' not in ecephys_mod.data_interfaces:
+                ecephys_mod.add_data_interface(FilteredEphys(name='Processed'))
```

Comment:
```suggestion
                ecephys_mod.add(FilteredEphys(name='Processed'))
```

Reply:
why this?
see https://pynwb.readthedocs.io/en/stable/pynwb.base.html#pynwb.base.ProcessingModule.add_data_interface

Reply:
Simpler. We haven't used `add_data_interface` in quite a while.